File: proselint/checks/annotations.py
"""
Annotation left in text.

---
layout:     post
source:     SublimeLinter-annotations
source_url: http://bit.ly/16Q7H41
title:      archaism
date:       2014-06-10 12:31:19
categories: writing
---

Annotation left in text.

"""

from proselint.registry.checks import Check, engine, types

check = Check(
    check_type=types.Existence(
        items=(
            "FIXME",
            "FIX ME",
            "TODO",
            "todo",
            "ERASE THIS",
            "FIX THIS",
        )
    ),
    path="annotations.misc",
    message="Annotation left in text.",
    engine=engine.Fast(opts=engine.RegexOptions(case_insensitive=False)),
)

__register__ = (check,)

File: proselint-rs/crates/proselint-checks/src/annotations.rs
use proselint_registry::checks::{Check, Padding, types::*};

pub const EXAMPLES_PASS: &[&str] = &[
	"Smoke phrase with nothing flagged.",
	"Add it to the to-do list.",
];
pub const EXAMPLES_FAIL: &[&str] = &[
	"Add it to the TODO list.",
];

const CHECK: Check = Check {
	check_type: &Existence {
		items: &[
			"FIXME",
			"FIX ME",
			"TODO",
			"todo",
			"ERASE THIS",
			"FIX THIS",
		],
		padding: Padding::WordsInText,
		exceptions: &[],
	},
	path: "annotations.misc",
	msg: "Annotation left in text.",
	..Check::default()
};

pub const REGISTER: &[Check] = &[CHECK];
